src/strategies/instances/compound-v3/CompoundV3StrategyFactory.sol

```diff
@@ -0,0 +1,181 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.22;
+
+import { IEarnVault } from "@balmy/earn-core/interfaces/IEarnStrategy.sol";
+import { StrategyId, StrategyIdConstants } from "@balmy/earn-core/types/StrategyId.sol";
+import { IEarnBalmyStrategy } from "../../../interfaces/IEarnBalmyStrategy.sol";
+import { IGlobalEarnRegistry } from "../../../interfaces/IGlobalEarnRegistry.sol";
+import { BaseStrategyFactory } from "../base/BaseStrategyFactory.sol";
+import { CompoundV3Strategy, IGlobalEarnRegistry, ICERC20, ICometRewards, IERC20 } from "./CompoundV3Strategy.sol";
+
+struct CompoundV3StrategyData {
+  IEarnVault earnVault;
+  IGlobalEarnRegistry globalRegistry;
+  address asset;
```

Comment:
I think we don't need to take the asset as params, since we can ask the `cToken` for the underlying